lib/exabgp/bgp/message/update/attribute/mprnlri.py

```diff
@@ -171,6 +174,10 @@ def unpack (cls, data, negotiated):
 				length,nlri = NLRI.unpack(afi,safi,data,addpath,'',IN.ANNOUNCED)
 				nlris.append(nlri)
 
+			if length == 0:
```

Comment:
Ok.  But at least something like the following would be in order:

if length == 0: raise Exception("sub-calls should not return length zero")

You don't want a bogus BGP route triggering a parsing bug itself triggering an endless loop, so it's nice to have a safeguard.